### ISSUE TITLE
Saving a tilemap also saves the tileset

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#5624]
Tested on Godot 2.0.3 and 2.0.4, Windows 10 64 bits

Sometimes when I save a tilemap, Godot also saves the tileset it is using, and as a side effect it produces unwanted diffs in Git.
I expected the editor  not to touch the tileset at all, because the tilemap is using it, it's not supposed to modify the tiles themselves.
This happens in my project for a while, but I never found how until now.

Here is a test project:
[TilemapSavesTileset.zip](https://github.com/godotengine/godot/files/356028/TilemapSavesTileset.zip)
Open it, and save the scene. Notice Godot tries to save the tileset:
`Also saving modified external resource: res://tilesets/terrain_tileset2.tres`

**Comment by MarianoGnu** [comment]
is this bad? when saving an scene are linked resources are saved too, this is expected :expressionless:

**Comment by vnen** [comment]
It saves but does not produce any diff, so it's not that harmful. Maybe save only what's changed? That'd likely increase performance a bit. It seems to be like that but only after you saved the scene once.

**Comment by akien-mga** [comment]
What's weird is the `saving modified external resource` when the resource hasn't been modified at all. Either the message is wrong, or there's a missing check somewhere.

**Comment by Zylann** [comment]
This is not very harmful, but annoying because it does two things wrong:
- It re-saves a resource that has no reason to be saved, it's like saving a texture when you save a Sprite node :s
- Sometimes, it produces diffs in Git (not always, some of it might have been fixed in b391306828294bc28675e135e7c2232ac6b8305a), or worse, if you use Perforce you'll have to checkout the file while you are not actually working on it.

**Comment by vnen** [comment]
Actually it does make sense to save external resources. E.g. if you edit a script and run the scene, you expect the scene and the script to be saved so you can test it. The same applies for any other external resource.

If the editor stop saving external resources, we will certainly see a lot of complaints about users losing work. Really, usability-wise it'll be worse, since you'll have to remember to save each and every resource in a scene (which can be a lot and scattered through many nodes).

The only problem I see here is the editor saving unmodified stuff.

**Comment by Zylann** [comment]
Ok I get it, it makes sense to save modifications on linked resources before saving the scene, for better UX. So yeah, unmodified stuff should not be saved again.

Tileset resources actually never have "edits", because they are most often exported from a tileset scene. There would be edits to save if it's edited from the inspector, but I never use that.
And when I edit a resource with the script editor or the inspector, I remember to save it anyways. So in the end, Godot doesn't have to save external resources when saving a scene that often. I was surprised it could. So if a scene uses an external resource I forget to save, it will save it anyways? Interesting :)

**Comment by MarianoGnu** [comment]
In fact i have edited tilesets using inspector to change some tiles.

Not sure of how it works, but probably a flag migth change this behaviour
https://i.gyazo.com/69bde17d2d71597e37cd3b65e45b5d31.png

**Comment by hubbyist** [comment]
Adding a flag like "read-only" for some external library type resources may be a solution. On first modification that godot must write to resource, a popup saying "will you modify this external resource from current scene?" confirmation box gives ability to continue working with resource without switching scenes.

**Comment by akien-mga** [comment]
Is it still valid in the current master branch?

**Comment by Zylann** [comment]
I cannot reproduce it on a from-scratch project.

**Comment by akien-mga** [comment]
Let's mark it as fixed then; feel free to reopen/comment or open a new issue if it pops up again.
